manual/tools/texquote2.ml

```diff
@@ -0,0 +1,143 @@
+type environment =
+  | Caml
+  | Verbatim
+  | Verbatim_like
+
+let (current_environment : environment option ref) = ref None
+
+let in_quotes = ref false
+
+let verbatim_end_in = ref ""
+
+let verbatim_end_out = ref ""
```

Comment:
Could remove these two global variables `verbatim_end_in`, `verbatim_end_out` and instead make them part of the `Verbatim` constructor: `Verbatim of string * string`.